### PR TITLE
Added Paging to msgraph-user-list command and Deprecating msgraph-user-get-delta

### PR DESCRIPTION
**Pull request by roysagi**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19939

## Description
msgraph-user-get-delta - added deprecation description to command.
msgraph-user-list - added pagination to command.
 
## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/blob/ms_graph_bug_fix/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml)
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/ms_graph_bug_fix/Integrations/MicrosoftGraphUser/CHANGELOG.md)

### CODE REVIEW

**Comment by roysagi** [comment]
@Arsenikr - FYI

File: Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml [hunk 1]
```diff
@@ -218,8 +218,10 @@ script:
       required: false
       secret: false
     deprecated: false
```
Comment (ronykoz) [inline Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml]: so this should probably be deprecated too

File: Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py [hunk 2]
```diff
@@ -349,17 +349,31 @@ def get_user(user, properties):
...
+    next_page = demisto.args().get('next_page', None)
+    users_data, result_next_page = list_users(properties, next_page)
     users_readable, users_outputs = parse_outputs(users_data)
     human_readable = tableToMarkdown(name='All Graph Users', t=users_readable, removeNull=True)
+
     outputs = {'MSGraphUser(val.ID == obj.ID)': users_outputs}
+    if result_next_page:
+        human_readable += "To get further results, enter this to the next_page parameter:\n" + str(result_next_page)
+        # .NextPage.indexOf(\'http\')>=0 : will make sure the NextPage token will always be updated because it's a url
+        outputs['MSGraphUser(val.NextPage.indexOf(\'http\')>=0)'] = {'NextPage': result_next_page}
```
Comment (ronykoz) [inline Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py]: i think that you should better use setIntegrationContext

File: Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py [hunk 3]
```diff
@@ -349,17 +349,31 @@ def get_user(user, properties):
...
-    users_data = list_users(properties)
-
+    next_page = demisto.args().get('next_page', None)
+    users_data, result_next_page = list_users(properties, next_page)
     users_readable, users_outputs = parse_outputs(users_data)
     human_readable = tableToMarkdown(name='All Graph Users', t=users_readable, removeNull=True)
+
     outputs = {'MSGraphUser(val.ID == obj.ID)': users_outputs}
+    if result_next_page:
+        human_readable += "To get further results, enter this to the next_page parameter:\n" + str(result_next_page)
```
Comment (ronykoz) [inline Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py]: Add a ScreenShot to the PR for how does it look in the warroom

Comment (ronykoz) [reply Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py]: and maybe just add this info in the metadata of the tabletomarkdown

**Comment by roysagi** [comment]
Here is how the NextPage token is printed to the user:
image

And this is how it appears in context:
image

File: Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py (hunk 3 above)
Comment (roysagi) [reply Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py]: done!

File: Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py [hunk 4]
```diff
@@ -349,17 +349,35 @@ def get_user(user, properties):
...
-
+    next_page = demisto.args().get('next_page', None)
+    users_data, result_next_page = list_users(properties, next_page)
     users_readable, users_outputs = parse_outputs(users_data)
     human_readable = tableToMarkdown(name='All Graph Users', t=users_readable, removeNull=True)
+
     outputs = {'MSGraphUser(val.ID == obj.ID)': users_outputs}
+
+    if result_next_page:
+        human_readable = tableToMarkdown(name='All Graph Users', t=users_readable, removeNull=True,
```
Comment (ronykoz) [inline Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py]: i would do
metadata="To get further results, enter this to the next_page parameter:\n" + str(result_next_page)

and then remove line 355 and put the human readable assignment in line 363 after the if

Comment (roysagi) [reply Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py]: ok

File: Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml [hunk 5]
```diff
@@ -317,6 +319,12 @@ script:
       name: properties
       required: false
       secret: false
+    - default: false
+      description: The URL for the next page in the list
```
Comment (kirbles19) [inline Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml]: Add a period.

File: Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml [hunk 6]
```diff
@@ -355,8 +363,13 @@ script:
     - contextPath: MSGraphUser.UserPrincipalName
       description: User's principal name.
       type: Unknown
+    - contextPath: MSGraphUser.NextPage
+      description: A token to be put to the next list command to get further results.
```
Comment (kirbles19) [inline Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml]: A token to pass to the next list command to retrieve additional results.

File: Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml [hunk 7]
```diff
@@ -217,9 +217,11 @@ script:
...
       required: false
       secret: false
-    deprecated: false
-    description: Gets newly created, updated, or deleted users without performing
-      a full read of the entire user collection.
+    deprecated: true
+    description: DEPRECATED  - We recommend not useing this command since it will
+      return a single page only, it is equivalent to msgraph-user-list command - Gets
+      newly created, updated, or deleted users without performing a full read of the
+      entire user collection.
```
Comment (kirbles19) [inline Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml]: Deprecated. This command only returns a single page. Use the msgraph-user-list command instead, which gets newly created, updated, or deleted users without performing a full read of the entire user collection.

**Comment by kirbles19** [comment]
@roysagi Done.

**Comment by roysagi** [comment]
@kirbles19 - great just making sure, have you looked at this messages that are being printed to the user as well? https://github.com/demisto/content/pull/4832#issuecomment-552422273

**Comment by roysagi** [comment]
> @kirbles19 - great just making sure, have you looked at this messages that are being printed to the user as well? [#4832 (comment)](https://github.com/demisto/content/pull/4832#issuecomment-552422273)

Just talked to @kirbles19 - he approved this section as well.

**Pull request merged**